crypto/src/main/scala/coop/rchain/crypto/hash/Blake2b256.scala

```diff
@@ -27,4 +31,15 @@ object Blake2b256 {
     }
   }
 
+  def hash(inputs: Seq[ByteVector]): Array[Byte] = {
+    val outStream = new DigestOutputStream(new Blake2bDigest(256))
+    synchronized {
```

Comment:
Why are you synchronizing here?

Reply:
That's one of question I have too. Usually hash functions shouldn't depend from any shared memory regions, and therefore don't need any synchronization. However, the original code with synchronized { } region was written by Eitan and passed reviews. Therefore I assumed that I don't know something about actual implementation or JVM, and copied that synchronization region.

Reply:
It's not needed. It can be removed from the other function as well.

Reply:
I believe that the `synchronized` in the original code was mistakenly copied from an upstream source where it was protecting a mutable variable.  I agree with @lukasz-golebiewski that it shouldn't be necessary here.

Reply:
That's good. Removed both.